src/analysis.rs

```diff
@@ -99,17 +99,27 @@ impl Cost {
         self.0 <= weight.saturating_mul(1000)
     }
 
-    /// Return whether the cost is within the budget of
-    /// the given script witness of a transaction input.
+    /// Return the budget of the given script witness of a transaction output.
     #[cfg(feature = "elements")]
-    pub fn is_budget_valid(&self, script_witness: &Vec<Vec<u8>>) -> bool {
+    fn get_budget(script_witness: &Vec<Vec<u8>>) -> u32 {
         let mut sink = io::sink();
         let witness_stack_serialized_len = script_witness
             .consensus_encode(&mut sink)
             .expect("writing to sink never fails");
-        let budget = witness_stack_serialized_len + 50;
-        // Cast safety: witness stack serialized length cannot be more than 2^32 - 51
-        self.less_equal_weight(budget as u32)
+        assert!(
+            witness_stack_serialized_len <= u32::MAX as usize, // cast safety: Assume 32-bit machine or higher
+            "Serialized witness stack must be shorter than 2^32"
+        );
+        let budget = witness_stack_serialized_len as u32; // cast safety: witness_stack_serialized_len <= u32::MAX
```

Comment:
In dfb7f0be6fc35608cfdc9f8cc0a3e423995198b2:

We should use `u32::try_from` (or `usize::try_into` or whatever) and `expect` this, rather than doing an explicit assert followed by a cast.